src/utils/common.ts

```diff
@@ -364,7 +374,10 @@ export function detectProtocolByOwner(network: string, address?: string) {
   return null;
 }
 
-export async function defaultAccountDiscovery(
+/**
+ * @returns {number} between -1 and n where -1 means there are no accounts found.
+ */
+export async function getLastNotEmptyAccountIndex(
```

Comment:
I kind of understand what you want to achieve, but this requires really complex TS custom typings which in my opinion are an overkill. Especially as the TypeScript itself types and comments stuff the same way as I did. For example here's a typing info for `Array.findIndex()` which returns `-1...n`:
![image](https://github.com/aeternity/superhero-wallet/assets/8217670/2130a8e4-54c0-4344-b7b5-cb3bee23026c)
You see the return value is typed as `number` but there is an additional comment about the possible values.

Reply:
Is it really complex?

The typescript document itself uses `number` but IMO we can do better. I am pushing this because it's being used at  crucial helper methods and it's important to indicate this for someone to use this in the future.

Reply:
There are 2 issues:

1. There is no native range type. There are some ways to do it (https://stackoverflow.com/questions/39494689/is-it-possible-to-restrict-number-to-a-certain-range) but it only makes sense for small ranges. When you try to make a large range this is what you get:
![image](https://github.com/aeternity/superhero-wallet/assets/8217670/709f2293-d0af-4ac1-9384-42143c8167aa)

2. There is no typing for the infinity. When you try to use `typeof Infinity` TS has no knowledge how to handle it and returns `never`. Take a look:
![image](https://github.com/aeternity/superhero-wallet/assets/8217670/0d9fe86c-2941-4365-9235-d4ede7d7ca17)

So in short: there is a reason why TS engine does not provide any range typings for native JS functions. Instead it uses `number`. And I don't know any programming language that makes this easy or even possible.

Reply:
https://github.com/microsoft/TypeScript/issues/54925
Yup, the issue is still open. We can resolve this.